browser/brave_tab_helpers.cc

```diff
@@ -120,17 +121,18 @@ void AttachTabHelpers(content::WebContents* web_contents) {
   brave_rewards::RewardsTabHelper::CreateForWebContents(web_contents);
 
 #if BUILDFLAG(ENABLE_AI_CHAT)
-  if (ai_chat::features::IsAIChatEnabled()) {
-    content::BrowserContext* context = web_contents->GetBrowserContext();
-    auto skus_service_getter = base::BindRepeating(
-        [](content::BrowserContext* context) {
-          return skus::SkusServiceFactory::GetForContext(context);
-        },
-        context);
-    ai_chat::AIChatTabHelper::CreateForWebContents(
-        web_contents,
-        g_brave_browser_process->process_misc_metrics()->ai_chat_metrics(),
-        skus_service_getter, g_browser_process->local_state());
+  if (ai_chat::features::IsAIChatEnabled() &&
+      !web_contents->GetBrowserContext()->IsTor()) {
+    auto* profile =
+        Profile::FromBrowserContext(web_contents->GetBrowserContext());
+    if (!profile->IsGuestSession()) {
+      auto skus_service = skus::SkusServiceFactory::GetForContext(
+          profile->GetOriginalProfile());
```

Comment:
Q. I'm a little bit confused when reading https://github.com/brave/brave-core/blob/bf8fdc60ca8b5adf0f86383390449884eeb88a20/browser/skus/skus_service_factory.cc#L67 line.

Is it okay to use Skus service in private window like this?

Reply:
fyi - this is not the right way to control this. Decisions about which profiles the skus service is available in should be in SkuServiceFactory (and I believe they are). We should just be doing a null check here